src/Modules/ExteriorPowers/FreeModules.jl

```diff
@@ -65,29 +65,27 @@ function exterior_power(F::FreeMod, p::Int; cached::Bool=true)
 
   # Set the variable names for printing
   orig_symb = String.(symbols(F))
-  new_symb = Symbol[]
-  if iszero(p)
-    new_symb = [Symbol("1")]
-  else
-    for ind in combinations(n, p)
-      symb_str = orig_symb[ind[1]]
-      for i in 2:p
-        symb_str = symb_str * (is_unicode_allowed() ? "∧" : "^") * orig_symb[ind[i]]
+  result.S = function _get_koszul_symbols()
+    new_symb = Symbol[]
+    if iszero(p)
+      new_symb = [Symbol("1")]
+    else
```

Comment:
How about handling the case `iszero(p)` outside of the function? (Not very important, feel free to ignore)
```suggestion
  if iszero(p)
    result.S = [Symbol("1")]
  else
    result.S = function _get_koszul_symbols()
      new_symb = Symbol[]
```